packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js

```diff
@@ -14,494 +15,519 @@ const TokenFactory = artifacts.require("TokenFactory");
-    );
-
-    // Withdraw from the expired liquidation and check that the liquidation is deleted.
-    await emp.withdrawLiquidation("0", sponsor1, { from: liquidator });
-    await client.update();
-    assert.deepStrictEqual([], client.getExpiredLiquidations().sort());
-  });
+const Convert = decimals => number => parseFixed(number.toString(), decimals).toString();
 
-  it("Returns disputed liquidations", async function() {
-    const liquidator = sponsor2;
-
-    await emp.create({ rawValue: toWei("150") }, { rawValue: toWei("100") }, { from: sponsor1 });
-    await syntheticToken.transfer(liquidator, toWei("100"), { from: sponsor1 });
-
-    // Create a new liquidation for account[0]'s position.
-    const { liquidationId } = await emp.createLiquidation.call(
-      sponsor1,
-      { rawValue: "0" },
-      { rawValue: toWei("9999999") },
-      { rawValue: toWei("100") },
-      unreachableDeadline,
-      { from: liquidator }
-    );
-    await emp.createLiquidation(
-      sponsor1,
-      { rawValue: "0" },
-      { rawValue: toWei("9999999") },
-      { rawValue: toWei("100") },
-      unreachableDeadline,
-      { from: liquidator }
-    );
-    await client.update();
-    const liquidations = client.getUndisputedLiquidations();
-    const liquidationTime = liquidations[0].liquidationTime;
-
-    // There should be no disputed liquidations initially.
-    assert.deepStrictEqual([], client.getDisputedLiquidations().sort());
-
-    // Dispute the liquidation and make sure it no longer shows up in the list.
-    // We need to advance the Oracle time forward to make `requestPrice` work.
-    await mockOracle.setCurrentTime(Number(await emp.getCurrentTime()) + 1);
-    await emp.dispute(liquidationId.toString(), sponsor1, { from: sponsor1 });
-    await client.update();
-
-    // The disputed liquidation should no longer show up as undisputed.
-    assert.deepStrictEqual(
-      [
-        {
-          sponsor: sponsor1,
-          id: "0",
-          state: "2",
-          liquidationTime: liquidationTime,
-          numTokens: toWei("100"),
-          liquidatedCollateral: toWei("150"),
-          lockedCollateral: toWei("150"),
-          liquidator: liquidator,
-          disputer: sponsor1
-        }
-      ],
-      client.getDisputedLiquidations().sort()
-    );
-    assert.deepStrictEqual([], client.getUndisputedLiquidations().sort());
-
-    // Force a price such that the dispute fails, and then
-    // withdraw from the unsuccessfully disputed liquidation and check that the liquidation is deleted.
-    const disputePrice = toWei("1.6");
-    await mockOracle.pushPrice(web3.utils.utf8ToHex("UMATEST"), liquidationTime, disputePrice);
-    await emp.withdrawLiquidation("0", sponsor1, { from: liquidator });
-    await client.update();
-    assert.deepStrictEqual([], client.getDisputedLiquidations().sort());
+contract("ExpiringMultiPartyClient.js", function(accounts) {
+  configs.forEach(({ collateralDecimals, tokenName }) => {
```

Comment:
As discussed offline, I'd change this from `forEach` to a `for(let x of X)` or `for(let x; x < len; x++)`. For each sometimes gives unexpected results for asynchronous code: https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop#37576787

Reply:
Is there any async code being called in this for each, though? Or are you just saying it's safer to preempt any issues?

Reply:
I have updated to your spect @nicholaspai